packages/aws-cdk-lib/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts

```diff
@@ -152,4 +174,29 @@ export class ApplicationLoadBalancedFargateService extends ApplicationLoadBalanc
       throw new Error(`${name}: Must be a non-negative integer; received ${value}`);
     }
   }
+
+  private validateContainerCpu(containerCpu?: number, cpu: number = 256) { // default value for cpu is 256
```

Comment:
where are these defaults defined ?

Reply:
Thanks @ren-yamanashi , I see  the default values are in fargate task definition and not under being set under `props.cpu` provided, i would prefer if we keep the check without default value as we might miss the update in case it changes in future as well.

Reply:
@shikha372 

Thank you for your comment.

In the current `validateContainerCpu` method, it compares the values of `containerCpu` and `cpu` and returns an error if containerCpu exceeds cpu.

As you pointed out, referencing the default value of `FargateServiceBaseProps.cpu` in `ApplicationLoadBalancedFargateService`’s `validateContainerCpu` could indeed lead to an issue where we might miss the update in case it changes in future as well.

However, avoiding the use of the default value of `FargateServiceBaseProps.cpu` within `ApplicationLoadBalancedFargateService` to address this concern could lead to the following problem:

#### `validateContainerCpu` method might not validate properly

If don’t reference the default value of `FargateServiceBaseProps.cpu`, the `cpu` param in `validateContainerCpu` would need to be optional.
(because `props.cpu` is optional)

```typescript
  private validateContainerCpu(containerCpu?: number, cpu?: number) {
    /**
     * ... Implementation of the method
     */
  }
```

In this scenario, it becomes challenging to validate cases where:

- `containerCpu` is set, but cpu is not.

In this cases, an error should be returned if `containerCpu` exceeds the default value of `FargateServiceBaseProps.cpu` (256), but without referencing the default value within the method, determining if an error should be returned becomes difficult.

---

### Possible Approaches for Resolution

Taking these considerations into account, I’ve outlined the following possible approaches. What could you think?

Note: this approaches would result in referencing the default value of `FargateServiceBaseProps.cpu` in `ApplicationLoadBalancedFargateService`, but I thought it could be more appropriate than the current implementation, so I am suggesting them.

#### Override props.cpu in `ApplicationLoadBalancedFargateServiceProps`

override `props.cpu` in `ApplicationLoadBalancedFargateServiceProps` and pass it to the `FargateTaskDefinition` argument within `ApplicationLoadBalancedFargateService`, as shown below:

```typescript
export interface ApplicationLoadBalancedFargateServiceProps extends ApplicationLoadBalancedServiceBaseProps, FargateServiceBaseProps {
  /**
   * ... some comments
   *
   * @default 256
   */
  readonly cpu?: number;

 /**
  * ... some props
  */
}

export class ApplicationLoadBalancedFargateService extends ApplicationLoadBalancedServiceBase {
  // ... some implementation
  constructor(scope: Construct, id: string, props: ApplicationLoadBalancedFargateServiceProps = {}) {
    // ... some implementation

      this.taskDefinition = new FargateTaskDefinition(this, 'TaskDef', {
        cpu: props.cpu ?? 256,
        // ... some props
      });
  }
}   
```

Reply:
Thank you @ren-yamanashi for defining the alternate detail approach, I would modify the current behaviour of the construct by setting this default, I believe that if the value of containerCpu and cpu don't align this would result in a failed deployment, if the intention is to fail fast before the deployment itself, can we add a warning instead of an error in validation so that in case these values change in the future it will not block the customer deployments.